txpipe/twopoint_fourier.py

```diff
@@ -116,28 +121,32 @@ def run(self):
         # fiducial theory C_ell, which is used in the deprojection calculation
         tracer_sacc = self.load_tracers(nbin_source, nbin_lens)
 
-        with self.open_input("fiducial_cosmology", wrapper=True) as f:
-            cosmo = f.to_ccl()
+        if self.config["compute_theory"]:
+            with self.open_input("fiducial_cosmology", wrapper=True) as f:
+                cosmo = f.to_ccl()
 
-        theory_ell = np.unique(np.geomspace(1, 3000, 100).astype(int))
-        theory_sacc = theory_3x2pt(cosmo,
-                                   tracer_sacc,
-                                   bias=self.config["b0"],
-                                   smooth=True,
-                                   ell_values=theory_ell
-        )
+            theory_ell = np.unique(np.geomspace(1, 3000, 100).astype(int))
+            theory_sacc = theory_3x2pt(cosmo,
+                                    tracer_sacc,
+                                    bias=self.config["b0"],
+                                    smooth=True,
+                                    ell_values=theory_ell
+            )
+        else:
+            theory_sacc = sacc.Sacc()
 
         # Get the complete list of calculations to be done,
         # for all the three spectra and all the bin pairs.
         # This will be used for parallelization.
         calcs = self.select_calculations(nbin_source, nbin_lens)
 
-        # Load in the per-bin auto-correlation noise levels and
-        # mean response values
-        # Note - this is currently unused, because we are using the noise
-        # maps instead of an analytic form, but that could change later
-        # so I will leave this here.
-        tomo_info = self.load_tomographic_quantities(nbin_source, nbin_lens, f_sky)
+        if not self.config["analytic_noise"]:
+            # Load in the per-bin auto-correlation noise levels and
+            # mean response values
+            # Note - this is currently unused, because we are using the noise
+            # maps instead of an analytic form, but that could change later
+            # so I will leave this here.
```

Comment:
Is this comment still correct? is the analytic form now implemented?

Reply:
referring to this change in the config options
>     do_shear_shear: False
>     do_shear_pos: False
>     do_pos_pos: True
>     compute_theory: False

Reply:
I think it's incorrect, and come to think of it it may actually have been for a while. Even before my changes, compute_noise_analytic didn't make use of the tomographic info loaded at this stage, so I wonder if it's left over from a while ago and is no longer relevant. Do you think it could be worth removing altogether?

Reply:
yes I think that makes sense